libraries/chain/include/eosio/chain/hotstuff.hpp

```diff
@@ -12,10 +12,46 @@ namespace eosio::chain {
 
    using hs_bitset = boost::dynamic_bitset<uint32_t>;
 
+   static digest_type get_digest_to_sign(const block_id_type& block_id, uint8_t phase_counter, const fc::sha256& final_on_qc){
```

Comment:
`static` => `inline`. Although I it seems this belongs in `hotstuff` namespace. Really I question why all this is not in `eosio::chain::hotstuff` namespace, but that is a bigger change we can make later.

Reply:
Moving all this to the hotstuff namespace (and probably moving hotstuff.hpp out of chain and into the hotstuff library entirely?) can be a new issue (maybe a fast-follow, as it would not affect functionality).